### ISSUE TITLE
Update to Lua 5.3.5

### ISSUE THREAD

**Issue by swaldhoer** [swaldhoer/native-lua#48]
Download: [Lua 5.3.5](https://www.lua.org/ftp/lua-5.3.5.tar.gz)

- [x] Update sources
- [x] Change build instructions for FreeBSD
- [x] Change build instructions for macOS  (nothing to do)

**Comment by swaldhoer** [comment]
```diff
diff --git a/Makefile5.3.4 b/Makefile5.3.5
index d71c75c..64c78f7 100644
--- a/Makefile5.3.4
+++ b/Makefile5.3.5
@@ -102,7 +102,7 @@ c89:


 freebsd:
-       $(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -lreadline"
+       $(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX -DLUA_USE_READLINE -I/usr/include/edit" SYSLIBS="-Wl,-E -ledit" CC="cc"

 generic: $(ALL)

@@ -110,7 +110,7 @@ linux:
        $(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"

 macosx:
-       $(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline" CC=cc
+       $(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline"

 mingw:
        $(MAKE) "LUA_A=lua53.dll" "LUA_T=lua.exe" \

```

**Comment by swaldhoer** [comment]
The diff of the `Makefile` shows a wrong setup for  FreeBSD. FreeBSD defines `LUA_USE_LINUX` and this implicitly redefines `LUA_USE_READLINE`:

`luaconf.h`

```c
#if defined(LUA_USE_LINUX)
#define LUA_USE_POSIX
#define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
#define LUA_USE_READLINE	/* needs some extra libraries */
#endif
```

**Comment by swaldhoer** [comment]
`LUA_USE_LINUX` defines `LUA_USE_DLOPEN`, and thus should need dl library (dynamic linking library). However, building with `LUA_USE_DLOPEN` and **not** specifying the dl library works (see
this 8cbc1b54d4fe9219193eef16e1e31b45cfa66eb2 or more precise this bf0ac2d0e1ea647731009c9fb6874217b7ef9f41)
Linking with dl library also works (see 314b97820dc9ce5bb34af31e385f04bc9bbf88c0).

**Comment by swaldhoer** [comment]
Running full tests on version `5.3.4` works locally (not on cirrus, not sure why). Running full tests fails in main.lua line 239 (https://github.com/swaldhoer/native-lua/runs/212079729).
Building the official release with make, leads to the same error, therefore there must be something wrong on the official build instructions.

**Comment by swaldhoer** [comment]
The Makefile bug is known https://www.lua.org/bugs.html#5.3.5-1.

**Comment by swaldhoer** [comment]
Fixed in #49
